org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -2013,43 +2020,90 @@ public void setNullableTypes(List<String> nullableTypes) {
 		this.nullableTypes = nullableTypes;
 	}
 
-	public boolean isAnnotationNullAnalysisEnabled() {
-		return !this.nonnullTypes.isEmpty() || !this.nullableTypes.isEmpty();
+	public void setNullAnalysisMode(FeatureStatus nullAnalysisMode) {
+		this.nullAnalysisMode = nullAnalysisMode;
+	}
+
+	public FeatureStatus getNullAnalysisMode() {
+		return this.nullAnalysisMode;
 	}
 
 	/**
+	 * update the null analysis options of all projects based on the null analysis mode
 	 * @return whether the options are changed or not
 	 */
 	public boolean updateAnnotationNullAnalysisOptions() {
+		switch (this.getNullAnalysisMode()) {
+			case automatic:
+				return this.updateAnnotationNullAnalysisOptions(true);
+			case interactive:
+				if (this.hasAnnotationNullAnalysisTypes()) {
+					String cmd = "java.compile.nullAnalysis.setMode";
```

Comment:
This command can be implemented via delegateCommand, and store it in workspace cache.